include/Dialect/LWE/IR/LWETypes.td

```diff
@@ -28,8 +28,10 @@ def LWECiphertext : LWE_Type<"LWECiphertext", "lwe_ciphertext"> {
   }];
 
   let parameters = (ins
-    LWE_BitFieldEncoding:$encoding,
-    LWE_LWEParams:$lwe_params
+  // FIXME: Encoding attrs belong on ops and types, should the encoding
+  //   attributes be a different class?
+    "::mlir::Attribute":$encoding,
```

Comment:
@j2kun curious if you there's an obvious answer here. I had created an attribute with a "AnyOf" constraint (any one of the LWE encoding attributes), but these attr's buildable types are "::mlir::Attribute" - I *think* in order to use them as type parameters they need a c++ buildable type.

I was looking into `AttrOrTypeParameters` but had some troubles using it, although I think I *should*. Does that seem like the right approach?

Reply:
As best I can tell the attribute constraints only work with ops. IMO it would be fine to have a generic Attribute parameter here, and dyn_cast it when we read it, since we will need to do that anyway if there are multiple options.

I guess the alternative is to have a base class with the width field somehow, though I'm not sure how to do that in tablegen, and not sure it's worth the effort since we'll need to cast either way.

Reply:
Yeah, doing the base class attribute would "make sense". I think Zama roughly achieved it for their unspecified secret key with custom C++ code, but that did have to use casting as well.